lib/Plack/Request.pm

```diff
@@ -132,6 +132,15 @@ sub headers {
     $self->{headers};
 }
 
+sub header_parameters {
+    my $self = shift;
+
+    return Hash::MultiValue->from_mixed(
+        map { $_ => +[ split /,\ */ => $self->headers->header($_) ] }
```

Comment:
This code is a bit confusing because you're calling `header()` in a list context, and *then* calls split on it. The PSGI spec requires them to be in a single value with comma separated, but still the line might be unintuitive.

I recommend you to write a unit test that tests the pattern of:

* `Foo: foo, bar, baz`
* `Foo: foo, bar` and `Foo: baz`